### DISCUSSION TITLE
Scaling to many concurrent visitors

### DISCUSSION THREAD

**Discussion by jmgirard** [surveydown-dev/surveydown#129] [Q&A]
Assuming the researcher is using free tiers of supabase and shinyapps, how well is surveydown expected to scale to many concurrent visitors/participants? If I recruit participants online (e.g., via Prolific) the normal way (i.e., without explicitly trying to stagger their recruitment), it would not be uncommon (in my experience) to have 10-30 concurrent participants at any time. Will this work out via the free tiers' number of workers, or will performance degrade or visitors be turned away? Or are you imagining that researchers will need to have the paid tiers or self-hosted servers to really scale?

**Comment by jhelvy** [comment]
Yeah we really need to do some heavier testing on this to know the answer. @pingfan-hu opened a simple survey in 30 tabs once to test it and it seemed to all run fine. But the only way to really know is to have a bunch of people concurrently take a survey and stress test it. The docs we've read seem to suggest that Supabase will be able to handle as much as 200 at once on the free tier. But shinyapps.io might struggle if you don't have enough RAM. The free tier only allows up to 1GB, so you may need to pay to get it up to 8GB. You could also try hosting it on other sites like Heroku. We linked to some [here](https://surveydown.org/deployment#deploying-to-other-hosting-services).

**Comment by jhelvy** [comment]
Added a [performance](https://surveydown.org/performance) page to the website.
